jvm/src/test/java/com/microsoft/bond/protocol/BinaryStreamReaderTest.java

```diff
@@ -0,0 +1,693 @@
+package com.microsoft.bond.protocol;
+
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.EOFException;
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+public class BinaryStreamReaderTest {
+
+    @Test
+    public void readInt8() {
+        // setup
+        byte[] inputBytes = new byte[]{
+                (byte) 0x5A,
+                (byte) 0xE3
+        };
+        ByteArrayInputStream bais = new ByteArrayInputStream(inputBytes);
+        BinaryStreamReader r = new BinaryStreamReader(bais);
+
+        // read first
+        try {
+            byte v = r.readInt8();
+            assertEquals((byte) 0x5A, v);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read second
+        try {
+            byte v = r.readInt8();
+            assertEquals((byte) 0xE3, v);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // end of stream
+        try {
+            byte v = r.readInt8();
+            fail("Operation can't succeed: " + v);
+        } catch (EOFException e) {
+            // success
+        } catch (IOException e) {
```

Comment:
We're using ByteArray stream, so it should never throw an IOException other than EOF. If the testee has a bug that throws it, we catch it here.

---
In reply to: [119250643](https://github.com/Microsoft/bond/pull/465#discussion_r119250643) [](ancestors = 119250643)